src/App.js

```diff
@@ -1,60 +1,99 @@
-import React from "react";
-import { onChildAdded, push, ref, set } from "firebase/database";
-import { database } from "./firebase";
+import React, { useState, useEffect } from "react";
+import { onChildAdded, push, ref as databaseRef, set } from "firebase/database";
+import { database, storage } from "./firebase";
+import {
+  getDownloadURL,
+  ref as storageRef,
+  uploadBytes,
+} from "firebase/storage";
 import logo from "./logo.png";
 import "./App.css";
+import Card from "react-bootstrap/Card";
 
 // Save the Firebase message folder name as a constant to avoid bugs due to misspelling
-const MESSAGE_FOLDER_NAME = "messages";
-
-class App extends React.Component {
-  constructor(props) {
-    super(props);
-    // Initialise empty messages array in state to keep local state in sync with Firebase
-    // When Firebase changes, update local state, which will update local UI
-    this.state = {
-      messages: [],
-    };
-  }
-
-  componentDidMount() {
-    const messagesRef = ref(database, MESSAGE_FOLDER_NAME);
+const POSTS_FOLDER_NAME = "posts";
+const IMAGES_FOLDER_NAME = "images";
+
+const App = () => {
+  const [posts, setPosts] = useState([]);
+  const [input, setInput] = useState("");
+  const [fileInput, setFileInput] = useState("");
+  const [fileInputFile, setFileInputFile] = useState(null);
+
+  const postRef = databaseRef(database, POSTS_FOLDER_NAME);
+
+  useEffect(() => {
     // onChildAdded will return data for every child at the reference and every subsequent new child
-    onChildAdded(messagesRef, (data) => {
+    const post = [];
+    onChildAdded(postRef, (data) => {
+      post.push({ key: data.key, val: data.val() });
       // Add the subsequent child to local component state, initialising a new array to trigger re-render
-      this.setState((state) => ({
-        // Store message key so we can use it as a key in our list items when rendering messages
-        messages: [...state.messages, { key: data.key, val: data.val() }],
-      }));
+      setPosts([...post]);
     });
-  }
-
+  }, []);
   // Note use of array fields syntax to avoid having to manually bind this method to the class
-  writeData = () => {
-    const messageListRef = ref(database, MESSAGE_FOLDER_NAME);
-    const newMessageRef = push(messageListRef);
-    set(newMessageRef, "abc");
+
+  const postListItems = posts.map((post) => (
+    <Card style={{ width: "18rem" }} key={post.key} className="post">
+      <Card.Img variant="top" src={post.val.imageLink} className="image" />
+      <Card.Body>
+        <Card.Text>
+          Post Date: {new Date().toLocaleString("en-GB", { timeZone: "Japan" })}
+        </Card.Text>
+        <Card.Text>{post.val.text}</Card.Text>
+      </Card.Body>
+    </Card>
+  ));
+
+  const handlePostChange = (e) => {
```

Comment:
if you are using this function, better rename it. It is not responsible for changing of posts, but actually handling change of a single input.